tests/components/image_processing/test_init.py

```diff
@@ -4,289 +4,231 @@
 import homeassistant.components.http as http
 import homeassistant.components.image_processing as ip
 from homeassistant.const import ATTR_ENTITY_PICTURE
-from homeassistant.core import callback
 from homeassistant.exceptions import HomeAssistantError
-from homeassistant.loader import DATA_CUSTOM_COMPONENTS
-from homeassistant.setup import setup_component
+from homeassistant.setup import async_setup_component
 
-from tests.common import (
-    assert_setup_component,
-    get_test_home_assistant,
-    get_test_instance_port,
-)
+from tests.common import assert_setup_component, async_capture_events
 from tests.components.image_processing import common
 
 
-class TestSetupImageProcessing:
-    """Test class for setup image processing."""
-
-    def setup_method(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-
-    def teardown_method(self):
-        """Stop everything that was started."""
-        self.hass.stop()
-
-    def test_setup_component(self):
-        """Set up demo platform on image_process component."""
-        config = {ip.DOMAIN: {"platform": "demo"}}
-
-        with assert_setup_component(1, ip.DOMAIN):
-            setup_component(self.hass, ip.DOMAIN, config)
+def get_url(hass):
+    """Return camera url."""
+    state = hass.states.get("camera.demo_camera")
+    return f"{hass.config.internal_url}{state.attributes.get(ATTR_ENTITY_PICTURE)}"
 
-    def test_setup_component_with_service(self):
-        """Set up demo platform on image_process component test service."""
-        config = {ip.DOMAIN: {"platform": "demo"}}
 
-        with assert_setup_component(1, ip.DOMAIN):
-            setup_component(self.hass, ip.DOMAIN, config)
-
-        assert self.hass.services.has_service(ip.DOMAIN, "scan")
+async def setup_image_processing(hass, aiohttp_unused_port, enable_custom_integrations):
```

Comment:
`enable_custom_integrations` is a fixture that just pops from hass.data. We don't need to pass it here.
```suggestion
async def setup_image_processing(hass, aiohttp_unused_port):
```

Reply:
Thanks!🤦